langchain/vectorstores/redisearch.py

```diff
@@ -0,0 +1,189 @@
+                        self.embedding_function(text), dtype=np.float32
+                    ).tobytes(),
+                    "metadata": json.dumps(metadata),
+                },
+            )
+            ids.append(key)
+        return ids
+
+    def similarity_search(
+        self, query: str, k: int = 4, **kwargs: Any
+    ) -> List[Document]:
+        # Creates embedding vector from user query
+        embedding = self.embedding_function(query)
+
+        # Prepare the Query
+        return_fields = ["metadata", "content", "vector_score"]
+        vector_field = "content_vector"
+        hybrid_fields = "*"
+        base_query = (
+            f"{hybrid_fields}=>[KNN {k} @{vector_field} $vector AS vector_score]"
+        )
+        redis_query = (
+            Query(base_query)
+            .return_fields(*return_fields)
+            .sort_by("vector_score")
+            .paging(0, k)
+            .dialect(2)
+        )
+        params_dict: Mapping[str, str] = {
+            "vector": str(np.array(embedding).astype(dtype=np.float32).tobytes())
+        }
+
+        # perform vector search
+        results = self.client.ft(self.index_name).search(redis_query, params_dict)
+
+        documents = [
+            Document(page_content=result.content, metadata=json.loads(result.metadata))
+            for result in results.docs
+        ]
+
+        return documents
+
+    @classmethod
+    def from_texts(
+        cls,
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        index_name: Optional[str] = None,
+        **kwargs: Any,
+    ) -> RediSearch:
+        """Construct RediSearch wrapper from raw documents.
+        This is a user-friendly interface that:
+            1. Embeds documents.
+            2. Creates a new index for the embeddings in the RediSearch instance.
+            3. Adds the documents to the newly created RediSearch index.
+        This is intended to be a quick way to get started.
+        Example:
+            .. code-block:: python
+                from langchain import RediSearch
+                from langchain.embeddings import OpenAIEmbeddings
+                embeddings = OpenAIEmbeddings()
+                redisearch = RediSearch.from_texts(
+                    texts,
+                    embeddings,
+                    redisearch_url="redis://username:password@localhost:6379"
+                )
+        """
+        redisearch_url = get_from_dict_or_env(
+            kwargs, "redisearch_url", "REDISEARCH_URL"
+        )
+        try:
+            import redis
+            from redis.commands.search.field import TextField, VectorField
+            from redis.commands.search.indexDefinition import IndexDefinition, IndexType
+        except ImportError:
+            raise ValueError(
+                "Could not import redis python package. "
+                "Please install it with `pip install redis`."
```

Comment:
show pinned version? might be difficult unless it's a set constant. Slight maintenance burden.

Reply:
I referred to other code under 'vectorstores', but no specific version was given in the error prompt.